src/spaced/spaced/game/player.hpp

```diff
@@ -10,21 +11,34 @@ class Player : public bave::IDrawable {
 
 	explicit Player(Services const& services);
 
+	void on_focus(bave::FocusChange const& focus_change);
 	void on_move(bave::PointerMove const& pointer_move);
 	void on_tap(bave::PointerTap const& pointer_tap);
 
-	void tick(bave::Seconds dt);
+	void tick(std::span<bave::NotNull<IDamageable*> const> targets, bave::Seconds dt);
 	void draw(bave::Shader& shader) const final;
 
-	void debug_stuff();
+	void inspect() {
+		if constexpr (bave::debug_v) { do_inspect(); }
+	}
 
 	void set_y(float y);
 	[[nodiscard]] auto get_y() const -> float { return ship.transform.position.y; }
 
 	bave::SpriteAnim ship{};
 
   private:
+	void do_inspect();
+	void debug_switch_weapon();
+
 	bave::Logger m_log{"Player"};
+	bave::NotNull<Services const*> m_services;
 	Controller m_controller;
+	std::unique_ptr<Weapon> m_weapon{};
+	std::vector<std::unique_ptr<Weapon::Round>> m_weapon_rounds{};
+
+	struct {
```

Comment:
why a struct here? just prefer to avoid raw `int` as a member?

Reply:
Just to contain all temp/debug stuff, currently there's only one thing (there were actually more before cleaning up and creating this PR).